src/plugins/core/conditional_format.ts

```diff
@@ -126,13 +188,11 @@ export class ConditionalFormatPlugin
   }
 
   adaptRanges(applyChange: ApplyRangeChange, sheetId?: UID) {
-    if (sheetId) {
+    const sheetIds = sheetId ? [sheetId] : Object.keys(this.cfRules);
+    for (const sheetId of sheetIds) {
       this.loopThroughRangesOfSheet(sheetId, applyChange);
-    } else {
-      for (const sheetId of Object.keys(this.cfRules)) {
-        this.loopThroughRangesOfSheet(sheetId, applyChange);
-      }
     }
+    this.loopThroughAllRanges(applyChange);
```

Comment:
looking at this code is very strange. You don't understand why you loop first over sheets and then a second time through all ranges.
I think `loopThroughRangesOfSheet` and `loopThroughAllRanges` should be renamed to reflect what they do